src/compound_targets.py

```diff
@@ -32,12 +32,18 @@ def cursorless_range(m) -> str:
     else:
         start = primitive_targets[0]
 
+    try:
+        range_type = m.cursorless_range_type
+    except AttributeError:
+        range_type = None
+
     return {
         "type": "range",
         "start": start,
         "end": primitive_targets[-1],
         "excludeStart": not is_anchor_included(range_connective),
         "excludeEnd": not is_active_included(range_connective),
+        "rangeType": range_type,
```

Comment:
Total nit, but I believe this will come through as `null`, which I think means it will appear when you record a test that doesn't use a vertical range.  It also doesn't quite match what's defined in typescript, which allows `undefined` here but not `null`.  When I want `undefined`, I'll usually save the map as a variable, then set the attribute in a try-catch.  See example here: https://github.com/pokey/cursorless-talon/blob/6acb3ffb6f3e17351cdc56ca638e3592ca724d68/src/modifiers/surrounding_pair.py#L43-L60

Reply:
I agreed with your changed and have pushed it. 
Regarding dfa it's not worse than what's already in develop but it's a a bit higher than what's in main. You're going to have to call this one.